src/target/linux/writemem.rs

```diff
@@ -0,0 +1,133 @@
+use nix::unistd::Pid;
+use std::{marker::PhantomData, mem};
+
+/// Allows to write data to different locations in debuggee's memory as a single operation.
+/// This implementation can select different strategies for different memory pages.
+pub struct WriteMemory<'a> {
+    pid: Pid,
+    write_ops: Vec<WriteOp>,
+    _marker: PhantomData<&'a mut ()>,
+}
+
+impl<'a> WriteMemory<'a> {
+    pub(super) fn new(pid: Pid) -> Self {
+        WriteMemory {
+            pid,
+            write_ops: Vec::new(),
+            _marker: PhantomData,
+        }
+    }
+
+    pub fn write<T>(mut self, val: &'a T, remote_base: usize) -> Self {
+        self.write_ops.push(WriteOp {
+            remote_base,
+            source_len: mem::size_of::<T>(),
```

Comment:
```suggestion
    pub fn write<T: ?Sized>(mut self, val: &'a T, remote_base: usize) -> Self {
        self.write_ops.push(WriteOp {
            remote_base,
            source_len: mem::size_of_val(val),
```

This allows passing in a slice to write.